File: js/기사단원의 무기.js
function solution(number, limit, power) {
    var answer = 0;
    let arr = Array.from({length: number+1}, () => 0);
    
    // 약수 개수 배열
    for (let i=1; i<number+1; i++) {
        for (let j=i; j<number+1; j+=i) {
            arr[j] += 1;
        }
    }

    for (let i=1; i<arr.length; i++) {
        if (arr[i] <= limit) {
            answer += arr[i];    
        } else {
            answer += power;
        }
    }
    return answer;
}

File: python/기사단원의 무기.py
# def solution(number, limit, power):
#     arr = [
#         i for i in range(1, number+1)
#     ]
#     arr2 = []
#     for a in arr:
#         cnt = 0
#         for i in range(1,int(a**0.5)+1):
#             if i * i == a:
#                 cnt += 1
#                 continue
#             if a % i == 0:
#                 cnt += 2
#         arr2.append(cnt)
#     answer =[]
#     for b in arr2:
#         if b <= limit:
#             answer.append(b)
#         else:
#             answer.append(power)
#     return sum(answer)

def solution(number, limit, power):
    # 약수 개수 배열
    arr = [0] * (number+1)
    
    # 효율적인 약수 계산
    for i in range(1, number+1):
        for j in range(i, number+1, i):
            arr[j] += 1
    
    answer = 0
    for elem in arr[1:]:
        if elem <= limit:
            answer += elem
        else:
            answer += power
            
    return answer